clicommand/pipeline_upload.go

```diff
@@ -613,24 +613,73 @@ func (cfg *PipelineUploadConfig) parseAndInterpolate(ctx context.Context, src st
 
 // gatherChangedFiles determines changed files in this build.
 func gatherChangedFiles(diffBase string) (mergeBase string, changedPaths []string, err error) {
-	// The --merge-base flag was only added to git-diff recently.
-	mergeBaseOut, err := exec.Command("git", "merge-base", diffBase, "HEAD").Output()
+	// Corporate needs you to find the differences between diffBase and HEAD.
+	diffBaseCommit, err := exec.Command("git", "rev-parse", diffBase).Output()
 	if err != nil {
-		return "", nil, gitMergeBaseError{diffBase: diffBase, wrapped: err}
+		return "", nil, gitRevParseError{arg: diffBase, wrapped: err}
 	}
-	mergeBase = strings.TrimSpace(string(mergeBaseOut))
-
-	gitDiff, err := exec.Command("git", "diff", "--name-only", mergeBase).Output()
+	headCommit, err := exec.Command("git", "rev-parse", "HEAD").Output()
 	if err != nil {
-		return mergeBase, nil, gitDiffError{mergeBase: mergeBase, wrapped: err}
+		return "", nil, gitRevParseError{arg: "HEAD", wrapped: err}
+	}
+	if strings.TrimSpace(string(diffBaseCommit)) == strings.TrimSpace(string(headCommit)) {
+		// They're the same commit, so `git diff` will (correctly) report no
+		// changes between them.
```

Comment:
It might be worth briefly document when this will happen.

Reply:
Done!